### PR TITLE
Add quarkus version to helm

### PR DESCRIPTION
**Pull request by aureamunoz**

### CODE REVIEW

**Comment by iocanel** [comment]
@aureamunoz @cmoulliard: This needs a rebase. When it's ready for review please remove the draft flag.

**Comment by aureamunoz** [comment]
Ok. It's still in progress...

File: locations/templates/quarkus-application/template.yaml [hunk 1]
```diff
@@ -330,13 +330,15 @@ spec:
           git_branch: main
           image_url: ${{ parameters.imageUrl }}
           appNamespace: ${{ parameters.namespace }}
+          version: ${{ parameters.version }}
+          quarkusVersion: ${{ parameters.quarkusVersion | extractVersionFromKey }}
```
Comment (cmoulliard) [inline locations/templates/quarkus-application/template.yaml]: We should remove the `| extractVersionFromKey` here as the `parameters.quarkusVersion` includes now the `major.minor.patch` version and not anymore `io.quarkus.platform:major.minor`

File: package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
     "node": "18 || 20"
   },
   "scripts": {
-    "dev": "concurrently \"yarn start --config ../../app-config.local.yaml\" \"NODE_OPTIONS=--no-node-snapshot yarn start-backend --config ../../app-config.local.yaml\"",
+    "dev": "concurrently \"yarn start --config ../../app-config.local.yaml\" \"NODE_OPTIONS=--no-node-snapshot yarn start-backend --inspect --config ../../app-config.local.yaml\"",
```
Comment (cmoulliard) [inline package.json]: That should be passed in another PR and ideally we should have (to be confirmed) 2 scripts; one to launch and another to debug

File: plugins/quarkus-backend/src/scaffolder/actions/quarkus/createApp.ts [hunk 3]
```diff
@@ -119,8 +119,11 @@ export const createQuarkusApp = () => {
             if (! ctx.input.values.starterCode) {
                 noCode = "true";
             }
+
+            const streamkey = buildStreamKeyFromVersion(ctx.input.values.quarkusVersion);
+
             const postData = {
-                streamKey: ctx.input.values.quarkusVersion ? ctx.input.values.quarkusVersion : 'io.quarkus.platform:3.9',
+                streamKey: streamkey ? streamkey : 'io.quarkus.platform:3.9',
```
Comment (cmoulliard) [inline plugins/quarkus-backend/src/scaffolder/actions/quarkus/createApp.ts]: Why is `3.9` hard coded ?

File: locations/templates/quarkus-application/template.yaml (hunk 1 above)
Comment (aureamunoz) [reply locations/templates/quarkus-application/template.yaml]: Yes, I missed this one

File: package.json (hunk 2 above)
Comment (aureamunoz) [reply package.json]: ok, I will remove it from here

File: plugins/quarkus-backend/src/scaffolder/actions/quarkus/createApp.ts (hunk 3 above)
Comment (aureamunoz) [reply plugins/quarkus-backend/src/scaffolder/actions/quarkus/createApp.ts]: I don't know, it was already hard coded previously to my changes, I've just kept it

Comment (cmoulliard) [reply plugins/quarkus-backend/src/scaffolder/actions/quarkus/createApp.ts]: We should perhaps throw an exception if the QuarkusVersion is not empty instead of hard a value that we know will change with future quarkus releases

Comment (aureamunoz) [reply plugins/quarkus-backend/src/scaffolder/actions/quarkus/createApp.ts]: Yes, it is indeed done [here](https://github.com/aureamunoz/backstage-playground/blob/f67696a46771ef894051f4261773413accc256d8/plugins/quarkus-backend/src/scaffolder/actions/quarkus/createApp.ts#L185), so the streamkey shouldn't be empty at this point. I will just removed the hard coded value

**Review by cmoulliard — APPROVED** [review]

**Comment by aureamunoz** [comment]
Could you please take a look on that in order to merge it @iocanel ?

**Review by iocanel — CHANGES_REQUESTED** [review]
Code looks great and works well too.

I think that we need to apply the change to all of our existing templates (not just the quarkus-application).

Additionally, we need to ensure that the value is quarkus-version label is properly set in the catalog-info.yaml too (currently appears as blank).

File: locations/templates/quarkus-application/manifests/helm/deploy/templates/_helpers.tpl [hunk 4]
```diff
@@ -65,6 +65,7 @@ Selector labels
 {{- define "quarkus-template.selectorLabels" -}}
 app.kubernetes.io/name: {{ include "quarkus-template.name" . }}
 app.kubernetes.io/instance: {{ .Release.Name }}
+app.quarkus.io/quarkus-version: {{ .Values.app.quarkusVersion }}
```
Comment (cmoulliard) [inline locations/templates/quarkus-application/manifests/helm/deploy/templates/_helpers.tpl:68]: Why did you added the Quarkus version as new selectorLabel ? If you add it here logically it should also be part of the labels too ?

Comment (aureamunoz) [reply locations/templates/quarkus-application/manifests/helm/deploy/templates/_helpers.tpl:68]: I don't know exactly why I added to the selectorLabels. Anyway these labels will be also part of the labels too, they are included by [this line](https://github.com/q-shift/backstage-playground/blob/f67696a46771ef894051f4261773413accc256d8/locations/templates/quarkus-application/manifests/helm/deploy/templates/_helpers.tpl#L55), right @iocanel ?

Comment (cmoulliard) [reply locations/templates/quarkus-application/manifests/helm/deploy/templates/_helpers.tpl:68]: Ok if they are also included within the labels. The selectorLabels are used to find a service, etc

**Comment by aureamunoz** [comment]
I applied the changes suggested by @iocanel, please review :-)

**Review by cmoulliard — APPROVED** [review]

**Comment by cmoulliard** [comment]
Can you fix the conflict please ? @aureamunoz

**Comment by aureamunoz** [comment]
> Can you fix the conflict please ? @aureamunoz

done

**Review by cmoulliard — APPROVED** [review]
